src/main/java/com/rickmorty/Controllers/UserController.java

```diff
@@ -0,0 +1,76 @@
+package com.rickmorty.Controllers;
+
+import com.rickmorty.Models.UserModel;
+import com.rickmorty.Services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    // Cria um novo usuário
+    @PostMapping
+    public ResponseEntity<?> createUser(@RequestBody UserModel user) {
+        if (userService.existsByEmail(user.getEmail())) {
+            return ResponseEntity
+                    .status(HttpStatus.BAD_REQUEST)
+                    .body(Map.of("error", "Email já cadastrado.", "code", 400));
+        }
+
+        LocalDate currentDate = LocalDate.now();
+        user.setDate_register(currentDate);
+        user.setDate_update(currentDate);
+
+        UserModel createdUser = userService.saveUser(user);
+        return new ResponseEntity<>(createdUser, HttpStatus.CREATED);
+    }
+
+    // Obtém todos os usuários
+    @GetMapping
+    public ResponseEntity<List<UserModel>> getAllUsers() {
+        List<UserModel> users = userService.getAllUsers();
+        return new ResponseEntity<>(users, HttpStatus.OK);
+    }
+
+
+    // Atualiza um usuário existente pelo e-mail
+    @PutMapping("/{email}")
+    public ResponseEntity<?> updateUserByEmail(@PathVariable String email, @RequestBody UserModel userDetails) {
+        try {
+            UserModel updatedUser = userService.updateUserByEmail(email, userDetails);
+            return new ResponseEntity<>(updatedUser, HttpStatus.OK);
+        } catch (RuntimeException e) {
+            return new ResponseEntity<>(Map.of("error", e.getMessage()), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    // GET para obter um usuário pelo e-mail
+    @GetMapping("/{email}")
+    public ResponseEntity<UserModel> getUserByEmail(@PathVariable String email) {
+        Optional<UserModel> userOptional = userService.getUserByEmail(email);
+        return userOptional.map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    // Exclui um usuário pelo e-mail (soft delete)
+    @DeleteMapping("/{email}")
```

Comment:
Remove os comentários e tudo o que for pertinente essa rota para que apaga usuários, tanto no controller quanto no service e repositório.

Reply:
O código foi devidamente ajustado.